core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -246,12 +246,21 @@ protected void add(DataFile file) {
 
   /** Add a delete file to the new snapshot. */
   protected void add(DeleteFile file) {
-    Preconditions.checkNotNull(file, "Invalid delete file: null");
```

Comment:
Why remove this check? Should this add it back in `add(DeleteFileHolder)`?

Reply:
The check moved to the `DeleteFileHolder` constructor, but is present in only one of two constructors now. It would be more explicit in `add`, I think.

Reply:
Yea initially tried to move it , but did it incomplete.  Added back to original location.